torch/_dynamo/output_graph.py

```diff
@@ -585,6 +585,16 @@ def compile_subgraph(
                     output.append(create_instruction("POP_TOP"))
             self.add_output_instructions(output + pass2.get_instructions())
 
+        # This is a hack to handle NumpyTensorVariable before returning it. This actually calls Tensor.numpy() to
+        # convert tensor to ndarray. Note that this doesn't handle TupleVariable so if more than one return values
+        # and one of them is NumpyTensorVariable this doesn't convert it and the result will be tensor instead of
+        # ndarray.
```

Comment:
Uhh, so did we not agree how to fix this? Is this going to go away before merging?

Reply:
Are you referring to the comment? If so I can make it better. I think we agreed on the approach, which is to add special logic to handle `NumpyTensorVariable` in `compile_subgraph()`.

Reply:
I'll update this